cirq-google/cirq_google/engine/processor_sampler.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2022 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
```

Comment:
Optional: should EngineSampler use this class underneath?

Reply:
I don't think it (easily) can.

`Engine.run_sweep` and `Processor.run_sweep` have meaningfully different semantics: The first will send out a job to one of potentially many processors. This logic happens server-side. While we could theoretically replicate it using a collection of Processors / ProcessorSamplers underneath EngineSampler, it wouldn't be a small change.

Reply:
I am starting to agree with maffoo.  It seems like ProcessorSampler is basically the same as a QuantumEngineSampler with a processor set already.   Seems like these two can/should be combined.

Reply:
After off-line conversations, I am fully on-board with this change.  

I think we should deprecate QuantumEngineSampler and switch engine.get_sampler(processor_id) to use engine.get_processor(processor_id).get_sampler() (or, alternatively, make QuantumEngineSampler a thin wrapper around this ProcessorSampler functionality).  Since this functionality currently supports multiple processor ids and sending an EngineProgram, it probably needs more thought and should be done in a follow-up PR.

Short story, I approve this PR as is.